app/service/user_filter.rb

```diff
@@ -0,0 +1,21 @@
+class UserFilter
+  def initialize(query:, group_id: nil)
+    @query = query
+    @group_id = group_id
+  end
+
+  def run
+    users - group_members
+  end
+
+  private
+
+  def users
+    @users ||= User.filter_by_email(@query).sort_by_email.limit(10)
+  end
+
+  def group_members
+    return [] unless @group_id
+    @group_members ||= User.joins(:groups_users).where('group_id = ?', @group_id)
```

Comment:
I actually started with something like this. I even had a `group` method that was loading the group. But the request was then running 3 queries (get users, get group, get users from group). I thought it was better to optimize it a bit and reduce to the only two queries we really need: getting users and getting the users belonging to a `group_id`. 

In the case of a non-existing group, the performances would be better with the third request to get the group, because it would mean we don't need to run the get members one. However, if the group exists (which should be the case most of the time when using this), we're saving a tiny bit of time by not querying the group.

I hope it makes sense. I don't mind changing it to query the group if you think it's better.